wgsl/index.bs

```diff
@@ -9286,6 +9286,247 @@ Any node which is not visited by these reachability analyses can be proven to be
+        _ = textureSample(t, s, pos.xy);
+      }
+    }
+  </xmp>
+</div>
+
+<div class=example1>
+  <figure>
+    <figcaption>Uniformity graph</figcaption>
+    <object type="image/svg+xml" data="img/uniformity_1.mmd.svg"></object>
+  </figure>
+</div>
+
+The example also shows that uniformity of the control flow after the if
+statement is the same as the uniformity prior to the if statement (CF_return
+being connected to CF_start).
+That is, the control flow is once again uniform after the if statement (because
+it is guaranteed to start as uniform control flow at the beginning of the entry
+point).
+If the `textureSample` function call been moved outside the if statement the
+program would have been valid.
+Likewise, if the condition of the if statement were a uniform value (e.g. each
+invocation read the same value from a [=uniform buffer=]), the program would
+also have been valid.
+
+#### Function-scope Variable Uniformity #### {#uniformity-example2}
+
+This example show both a valid and an invalid
+[[#sync-builtin-functions|barrier]] function call that depend on the value of a
+function-scope variable.
+The `workgroupBarrier` is invalid because the value of `x` is derived from the
+mutable module-scope variable `a`.
+The `storageBarrier` is valid because the value of `x` is derived from the
+immutable module-scope variable `b`. 
+This example highlights the [[#func-var-value-analysis|value analysis']]
+ability to separate different periods of uniformity in a function-scope
+variable's lifetime.
+This example also clearly shows that uniformity returns to the control flow
+after the first [=statement/if=] statement as that section of the graph is
+independent from the second if statement.
+
+<div class='example wgsl' heading='WGSL using function variable'>
+  <xmp highlight='rust'>
+    @group(0) @binding(0) var<storage, read_write> a : i32;
+    @group(0) @binding(1) var<uniform> b : i32;
+
+    @compute @workgroup_size(16,1,1)
+    fn main() {
+      var x : i32;
+      x = a;
+      if x > 0 {
+        // Invalid barrier function call.
+        workgroupBarrier();
+      }
+      x = b;
+      if x < 0 {
+        // Valid barrier function call.
+        storageBarrier();
+      }
+    }
+  </xmp>
+</div>
+
+<div class=example2>
+  <figure>
+    <figcaption>Uniformity graph</figcaption>
+    <object type="image/svg+xml" data="img/uniformity_2.mmd.svg"></object>
+  </figure>
+</div>
+
+Note: The subgraphs are only included in the example for ease of understanding.
+
+
+#### Composite Value Analysis Limitations #### {#uniformity-example3}
+
+One limitation of the uniformity analysis is that it does not track the
+[=components=] of a [=composite=] value independently.
+That is, any non-uniform component value will cause the analysis to treat the
+entire composite value as non-uniform.
+This example illustrates this issue and a potential workaround that shader
+authors can employ to avoid this limitation.
+
```

Comment:
No request for change.

But this is going to break the common pattern of making a single struct to collect all pipeline inputs.  I'm sure the committee will hear feedback.